### PR TITLE
feat: Enable advanced configuration for tolerance

### PR DESCRIPTION
**Pull request by mayitbeegh**

### CODE REVIEW

**Comment by mayitbeegh** [comment]
Screen Shot 2020-01-23 at 11 03 24 AM
Screen Shot 2020-01-23 at 11 03 47 AM
Screen Shot 2020-01-23 at 11 04 03 AM
Screen Shot 2020-01-23 at 11 04 06 AM

File: packages/client/src/components/config/AbstractMetricModal.tsx [hunk 1]
```diff
@@ -430,6 +548,124 @@ export abstract class AbstractMetricModal<T extends CanaryMetricSetQueryConfig>
...
                 {/* Inject the metric source specific JSX here */}
                 {this.getMetricSourceSpecificJsx()}
+                <Accordion
+                    allowZeroExpanded={true}
+                    onChange={() => this.setState({showAdvancedConfiguration : !this.state.showAdvancedConfiguration})}
+                >
+                    <AccordionItem>
+                      <AccordionItemHeading>
+                        <AccordionItemButton>
+                          <div className="titled-section-title">Advanced<FontAwesomeIcon className="img chevron" size="sm" color="black" icon={this.state.showAdvancedConfiguration?faChevronCircleDown:faChevronCircleRight} /></div>
```
Comment (melanahammel) [inline packages/client/src/components/config/AbstractMetricModal.tsx]: Some additional whitespace could make this line more readable
For example:

> <div className="titled-section-title">
          Advanced<FontAwesomeIcon className="img chevron" size="sm" color="black" 
                 icon={this.state.showAdvancedConfiguration ? faChevronCircleDown : faChevronCircleRight} />
</div>

File: packages/client/src/components/config/AbstractMetricModal.tsx [hunk 2]
```diff
@@ -430,6 +548,124 @@ export abstract class AbstractMetricModal<T extends CanaryMetricSetQueryConfig>
...
+                        </AccordionItemButton>
+                      </AccordionItemHeading>
+                      <AccordionItemPanel>
+                        <Alert variant="warning">For advanced users only. The following advanced features are incubating changes and are subject to breaking changes without notice in the future. Changing these values may affect your canary result in unexpected ways.</Alert>
+                      </AccordionItemPanel>
+                      <AccordionItemPanel>
+                        <TitledSection title="Effect Size" additionalClassname="effect-size" tooltipHeader="Effect Size" tooltipText={<>Controls the degree of statistical significance the metric needs to fail or fail critically. Metrics marked as critical can also define criticalIncrease and criticalDecrease.</>}/>
```
Comment (melanahammel) [inline packages/client/src/components/config/AbstractMetricModal.tsx]: for future reference: `tooltipText` can also just be a string

File: packages/client/src/validation/configValidators.ts [hunk 3]
```diff
@@ -39,7 +39,26 @@ const getCanaryMetricConfigSchema = (metricQueryObjectSchema: KvMap<Schema<any>>
               .required(),
             nanStrategy: mixed().oneOf(['remove', 'replace']),
             critical: boolean(),
-            mustHaveData: boolean()
+            mustHaveData: boolean(),
+            effectSize: object()
+                .shape({
+                  allowedIncrease: number()
+                      .notRequired(),
```
Comment (melanahammel) [inline packages/client/src/validation/configValidators.ts]: `notRequired()` is implicit, but if you want to be explicit you can leave it in

File: packages/client/src/components/config/AbstractMetricModal.tsx (hunk 1 above)
Comment (melanahammel) [reply packages/client/src/components/config/AbstractMetricModal.tsx]: `yarn format` took care of this

**Pull request merged**